src/main/java/spoon/support/visitor/equals/EqualsVisitor.java

```diff
@@ -41,5 +44,60 @@ protected void enter(CtElement e) {
 			fail();
 		}
 	}
+	protected boolean isNotEqual = false;
+
+	@Override
+	protected boolean biScan(CtRole role, Collection<? extends CtElement> elements, Collection<? extends CtElement> others) {
+		if (isNotEqual) {
+			return isNotEqual;
+		}
+		if (elements == null) {
+			if (others != null) {
+				return fail();
+			}
+			return isNotEqual;
+		} else if (others == null) {
+			return fail();
+		}
+		if ((elements.size()) != (others.size())) {
+			return fail();
+		}
+		super.biScan(role, elements, others);
+		return isNotEqual;
+	}
+
+	@Override
+	public boolean biScan(CtElement element, CtElement other) {
+		if (isNotEqual) {
+			return isNotEqual;
+		}
+		if (element == null) {
+			if (other != null) {
+				return fail();
+			}
+			return isNotEqual;
+		} else if (other == null) {
+			return fail();
+		}
+		if (element == other) {
+			return isNotEqual;
+		}
+
+		stack.push(other);
```

Comment:
Could we move `stack.push(other);` and `stack,pop()` somewhere into `CtBiScannerDefault`? I looks like implementation of `CtBiScannerDefault` in this PR is useless without calls of `stack.push(other);` and `stack,pop()`. Because `CtBiScannerDefault` calls `stack.peek()`, but nobody pushes data there.
WDYT?

Reply:
Good point. Done.